### PR TITLE
Adding support for msys.

### PR DESCRIPTION
**Pull request by nicolasnoble**

### CODE REVIEW

File: src/core/lib/iomgr/tcp_server_windows.c [hunk 1]
```diff
@@ -510,30 +510,17 @@ int grpc_tcp_server_add_port(grpc_tcp_server *s, const void *addr,
...
-    return 1;
-  } else {
-    return 0;
-  }
+  (void)s;
+  (void)port_index;
+  abort();
 }
 
 int grpc_tcp_server_port_fd(grpc_tcp_server *s, unsigned port_index,
```
Comment (ctiller) [inline src/core/lib/iomgr/tcp_server_windows.c]: Can we just remove these?
They shouldn't be here

Comment (nicolasnoble) [inline src/core/lib/iomgr/tcp_server_windows.c]: Sure.

**Comment by nicolasnoble** [comment]
Also, I've tested it locally against both msys and mingw.

**Comment by nicolasnoble** [comment]
And that seems to break something on Visual Studio. I'll investigate.

**Comment by nicolasnoble** [comment]
So the Visual Studio problem only happens with 2013. Joy. Let's see...

**Comment by ctiller** [comment]
Structurally LGTM, dgq - can you take a closer peek?

**Comment by dgquintas** [comment]
ack, sure

File: include/grpc/impl/codegen/port_platform.h [hunk 2]
```diff
@@ -82,36 +82,38 @@
    things.  */
 
 #if !defined(GPR_NO_AUTODETECT_PLATFORM)
+#if defined(_WIN64) || defined(WIN64) || defined(_WIN32) || defined(WIN32)
```
Comment (dgquintas) [inline include/grpc/impl/codegen/port_platform.h]: this line supersedes the following one

File: src/core/lib/support/tmpfile_msys.c [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  FILE *result = NULL;
+  char tmp_filename[MAX_PATH];
+  UINT success;
+
+  if (tmp_filename_out != NULL) *tmp_filename_out = NULL;
+
+  /* Generate a unique filename with our template + temporary path. */
+  success = GetTempFileNameA(".", prefix, 0, tmp_filename);
+  fprintf(stderr, "success = %d\n", success);
+  if (!success) goto end;
```
Comment (dgquintas) [inline src/core/lib/support/tmpfile_msys.c]: the use of `goto` here seems unjustified. How about
```
if (success) {
  /* Open a file there. */
  result = fopen(tmp_filename, "wb+");
  fprintf(stderr, "result = %p\n", result);
} 
if (result != NULL && tmp_filename_out) {
  *tmp_filename_out = gpr_strdup(tmp_filename);
}

return result;
```

File: include/grpc/impl/codegen/port_platform.h (hunk 2 above)
Comment (nicolasnoble) [inline include/grpc/impl/codegen/port_platform.h]: Not really ? The following line is a specialization to do the detection 32/64 bits, whereas this one is to detect if we're compiling for Windows at all.

Comment (dgquintas) [inline include/grpc/impl/codegen/port_platform.h]: Ah yes. 86 => 85, but 85 =/=> 86. Nevermind.

**Comment by nicolasnoble** [comment]
Fixes #5998.

**Comment by nicolasnoble** [comment]
Fixes #3254.

File: src/core/lib/support/tmpfile_msys.c (hunk 3 above)
Comment (nicolasnoble) [inline src/core/lib/support/tmpfile_msys.c]: It was really a copy from the windows version to begin with. But, sure.

Comment (nicolasnoble) [inline src/core/lib/support/tmpfile_msys.c]: Done.

**Comment by dgquintas** [comment]
LGTM

**Pull request merged**